docs/why.qmd

```diff
@@ -2,24 +2,224 @@
 title: "Why Ibis?"
 ---
 
+Ibis defines a Python dataframe API that supports any query engine -- the
+frontend for any backend data platform. This includes 20+ backends today, most
```

Comment:
nit - more precise words:
frontend -> "language frontend" 
backend data platform -> data processing engine
remove "most SQL-based" - because it does not add value to this statement.
excellent performance -> flexibility or adaptability into a composable data architecture

Reply:
reworded slightly but want to keep this fairly simple for a first-time reader